flink-runtime/src/main/java/org/apache/flink/runtime/concurrent/FutureUtils.java

```diff
@@ -807,20 +805,8 @@ public static Time toTime(FiniteDuration finiteDuration) {
 	 * @return Java 8 CompletableFuture
 	 */
 	public static <T, U extends T> CompletableFuture<T> toJava(Future<U> scalaFuture) {
-		final CompletableFuture<T> result = new CompletableFuture<>();
-
-		scalaFuture.onComplete(new OnComplete<U>() {
-			@Override
-			public void onComplete(Throwable failure, U success) {
-				if (failure != null) {
-					result.completeExceptionally(failure);
-				} else {
-					result.complete(success);
-				}
-			}
-		}, Executors.directExecutionContext());
-
-		return result;
+		//noinspection unchecked
+		return scala.compat.java8.FutureConverters.toJava((Future<T>) scalaFuture).toCompletableFuture();
```

Comment:
Where does this dependency come from? I fear that this is only a transitive dependency on which we should not directly depend.

Reply:
Thank Till for your review!

It is a dependency comes from org.scala-lang.modules as `org.scala-lang.modules:scala-java8-compat_2.11:0.7.0`, shipped within scala-lang and also mentioned in our `NOTICE` file.

Besides, this module is marked as "currently still experimental" and thus I might give up this hotfix. I had expected to reduce a custom implementation DirectExecutionContext and maintain it.